src/torrent.cpp

```diff
@@ -1810,7 +1810,7 @@ namespace libtorrent
 				have_pieces.push_back(p.index);
 			}
 
-			for (auto const i : have_pieces)
```

Comment:
the patch against clang that added this warning suggests that removing ``const`` here would also silence the warning. I think that's preferable. It's unfortunate, though, that it seems to discriminate primitive and other types.